src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -33,6 +34,55 @@ public static void UpdateFrameworkTypes(this IEnumerable<Schema> allSchemas)
             }
         }
 
+        public static void ReorderOperationParameters(this CodeModel codeModel)
```

Comment:
How this method correlates with https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/Models/MgmtRestClientBuilder.cs#L57 ?

Reply:
This method is moved from MgmtOutputLibrary to here, it is not a new method.
This method only sort the path parameters (which are all required) in a order as they appears in the path. We need this because the code model will just create the parameter in the order of definition.
And this method is changing the things in code model, therefore this runs like this:
1. First we modify the code model to make sure the path parameters are in right order.
2. Then we reach the function `BuildParameters` sort by required parameters, body parameter and then optional parameters

Reply:
It feels like it duplicates logic that we have in LLC: https://github.com/Azure/autorest.csharp/blob/0a87c4af88516fb49c3192b55b2aa5f02e6f9011/src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs#L210

Reply:
Maybe, but since this function in mgmt plane directly changed the code model, the changes will apply to RestOperations and every method in the generated client (the `AutoRest.CSharp.Input.Operation` class). Will this function in LLC do the same?
And this code in mgmt plane was introduced a long time ago, maybe we should refactor this later. I created an issue for this: https://github.com/Azure/autorest.csharp/issues/1985